### PR TITLE
Coverage

### LINKED ISSUE #129: Fix add_attributes() test
That test is not being execute (and would fail if so due to a typo):

https://github.com/pyugrid/pyugrid/blob/95f8ab8c7ccd2d7edc6bc586f62536aac60712ed/test/test_uvar.py#L60-L68

I have a fix here, but I am waiting for a few PRs to get merged to avoid conflicts.

### PR DESCRIPTION
**Pull request by ocefpaf**
Add coverage test and fixes #129

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,14 +14,14 @@ before_install:
     - conda update --all --yes
     - travis_retry conda create --yes -n TEST $CONDA --file requirements.txt
     - source activate TEST
-    - travis_retry conda install --yes --file requirements-dev.txt
+    - travis_retry pip install -r requirements-dev.txt
```
Comment (ocefpaf) [inline .travis.yml]: Note to self:  `pytest-cov` is unavailable in the default channel and outdated in the IOOS channel. For now it is OK to install the light weight dependencies (`requirements-dev.txt`) with `pip` and the heavy weight (`netCDF4-python`, `scipy`, etc) with `conda`.

The long term goal is to add those packages to conda-forge and just add that channel here.

**Comment by ocefpaf** [comment]
@ChrisBarker-NOAA first goal (a low hanging fruit I admit :grin:) from #22.  Can you review and merge?

File: .travis.yml (hunk 1 above)
Comment (ChrisBarker-NOAA) [inline .travis.yml]: I've got pytest-cov in the NOAA-ORR-ERD channel, if that helps.

CHB

On Feb 15, 2016, at 11:10 AM, Filipe <notifications@github.com> wrote:

In .travis.yml
<https://github.com/pyugrid/pyugrid/pull/130#discussion_r52934329>:

> @@ -14,14 +14,14 @@ before_install:
>      - conda update --all --yes
>      - travis_retry conda create --yes -n TEST $CONDA --file requirements.txt
>      - source activate TEST
> -    - travis_retry conda install --yes --file requirements-dev.txt
> +    - travis_retry pip install -r requirements-dev.txt

Note to self: pytest-cov is unavailable in the default channel and outdated
in the IOOS channel. For now it is OK to install the light weight
dependencies (requirements-dev.txt) with pip and the heavy weight (
netCDF4-python, scipy, etc) with conda.

The long term goal is to add those packages to conda-forge and just add
that channel here.

—
Reply to this email directly or view it on GitHub
<https://github.com/pyugrid/pyugrid/pull/130/files#r52934329>.

File: .travis.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ before_install:
     - conda update --all --yes
     - travis_retry conda create --yes -n TEST python=$TRAVIS_PYTHON_VERSION --file requirements.txt
     - source activate TEST
-    - travis_retry conda install --yes --file requirements-dev.txt
+    - travis_retry pip install -r requirements-dev.txt
```
Comment (ChrisBarker-NOAA) [inline .travis.yml]: are we not relying on conda anymore? I didn't think all the deve requirements were pip-installable?

**Comment by ChrisBarker-NOAA** [comment]
A bit confused about the pip for installing requirements, but the tests are working, so good to go.

File: .travis.yml (hunk 2 above)
Comment (ocefpaf) [inline .travis.yml]: Right now we rely on conda for the heavy weight dependencies listed in `requirements.txt`.  As soon as I get some time I will package the light weight (pure python and to easy pip install) from `requirements-dev.txt` on `conda-forge` and we can use a conda only solution.  In the end it does not matter much for Travis-CI as coda handle `pip installs` of pure package without a problem.

Comment (ChrisBarker-NOAA) [inline .travis.yml]: OK, makes sense -- I still see pip and conda get tangled up in dependencies, but for running tests on a CI, a conda install followed by a pip install is fine.

BTW, maybe a conda_requirements.txt and a requirements.txt (for pip)? they are mostly the same, but not always (sigh.)

File: .travis.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ install:
 
 script:
     - if [[ $TEST_TARGET == 'default' ]]; then
-        python setup.py test -a "--verbose";
+        python setup.py test -a '--verbose --cov=pyugrid';
```
Comment (ChrisBarker-NOAA) [inline .travis.yml]: so what happens if you don't have 100% coverage? I'd like to see the report, but we don't want that to be the equivalent of a test failure.

Comment (ocefpaf) [inline .travis.yml]: The report is at the end of the log. We can add a badge using those
coverage services, but I don't really like those.

File: .travis.yml (hunk 2 above)
Comment (ocefpaf) [inline .travis.yml]: > BTW, maybe a conda_requirements.txt and a requirements.txt (for pip)?
they are mostly the same, but not always (sigh.)

Here the packages names are compatible, so let's keep it simple and keep
only one requirement for both.

Comment (ChrisBarker-NOAA) [inline .travis.yml]: OK -- I guess wait and see if they diverge at some point, and deal with it then.

**Pull request merged**